pulp_ansible/app/models.py

```diff
@@ -211,6 +222,28 @@ class Meta:
         unique_together = ("pubkey_fingerprint", "signed_collection")
 
 
+class DownloadLog(BaseModel):
+    """
+    A download log for content units by user, IP and org_id.
+    """
+
+    content_unit = models.ForeignKey(
+        Content, on_delete=models.CASCADE, related_name="download_logs"
+    )
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="download_logs",
+    )
+    ip = models.GenericIPAddressField()
+    extra_data = models.JSONField(null=True)
+    user_agent = models.TextField()
+    repository = models.ForeignKey(
```

Comment:
Would repository version be more useful?

Reply:
Repository versions might be a bit more volatile if you are using `retain_repo_versions`. But if you like, you can get both.